src/engine/engine.rs

```diff
@@ -26,8 +27,26 @@ pub trait Filesystem: HasName + HasUuid {
 }
 
 pub trait BlockDev: HasUuid {
-    /// path of the device node
+    /// Get the path of the device node for this device.
     fn devnode(&self) -> PathBuf;
+
+    /// Get the user-settable string associated with this blockdev.
+    fn user_id(&self) -> &Option<String>;
+
+    /// Set the user-settable string associated with this blockdev.
```

Comment:
I don't understand why the only argument is an Option type.

Reply:
Trying to make it possible to unset the value.